src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -33,31 +34,65 @@
 
 public class JvmMonitor extends AbstractMonitor
 {
-  /**
+  /*
    * The following GC-related code is partially based on
    * https://github.com/aragozin/jvm-tools/blob/e0e37692648951440aa1a4ea5046261cb360df70/
    * sjk-core/src/main/java/org/gridkit/jvmtool/PerfCounterGcCpuUsageMonitor.java
    */
 
   private enum GcGeneration
   {
-    YOUNG(GarbageCollectors.youngName(), 0), OLD(GarbageCollectors.oldName(), 1);
+    YOUNG(0) {
+      @Override
+      String readableGcName(String name)
+      {
+        if (name.contains("Copy")) {
```

Comment:
They are not clear. For example all old GCs except G1 have "MarkSweep" part in their names, but only one of them is CMS.